src/common/components/fix-instruction-processor.tsx

```diff
@@ -24,62 +25,144 @@ export class FixInstructionProcessor {
         'i',
     );
 
-    public process(fixInstruction: string): JSX.Element {
+    private readonly foregroundRecommendedColorText = 'Use foreground color: ';
+    private readonly foregroundRecommendedRegExp = new RegExp(
+        `${this.foregroundRecommendedColorText}${this.colorValueMatcher}`,
+        'i',
+    );
+    private readonly backgroundRecommendedColorText = 'Use background color: ';
+    private readonly backgroundRecommendedRegExp = new RegExp(
+        `${this.backgroundRecommendedColorText}${this.colorValueMatcher}`,
+        'i',
+    );
+
+    private readonly contrastRatioText = 'Expected contrast ratio of ';
+    private readonly contrastRatioRegExp = new RegExp(`${this.contrastRatioText}`, 'i');
+    private readonly originalMiddleSentence = ' and the original foreground color: ';
+
+    private readonly recommendEndSentence = ' to meet a contrast ratio of #.##:1.';
+
+    public process(fixInstruction: string, recommendation: RecommendColor): JSX.Element {
+        fixInstruction += '.';
```

Comment:
This extra period is now going on the end of all how to fix instructions, not just color contrast ones like you probably intended. You'll want to move it so that it's only added when the instruction is updated to need it. That's causing most of the [report test failures here](https://dev.azure.com/accessibility-insights/accessibility-insights-web/_build/results?buildId=22084&view=logs&j=334911b1-973e-5a6d-17dc-0139ae99369b&t=cfac4a3e-83e7-50bf-eb8a-abbf08395129).